### DISCUSSION TITLE
A20PLUS options missing

### DISCUSSION THREAD

**Discussion by ranatalus** [noahm/DDRCardDraw#48] [General]
I am running a tournament in October, and I've identified a few things that need to be updated:
Jackets for new songs missing _(in progress by me)_
Difficulty changes and new songs added to a20plus.json _(in progress by a friend)_
Song exclusion options (omit Gold Cab exclusive, Extra Exclusive, etc) - _needs to be taken_

Anyone up for fixing the song exclusion options and we'll get the rest taken care of and merged?

**Comment by noahm** [comment]
Sure I can make a pass across the latest locked and unlocked songs to get that updated!

**Comment by noahm** [comment]
I spent a while upgrading the script for scraping A20plus song data from ZiV and it's now giving us fully up to date song info (with help from SkillAttack to correct the difficulty levels where ZiV has it wrong). See https://github.com/noahm/DDRCardDraw/pull/49 for the work in progress.

I'll go track down and try to update the current set of exclusives next.

**Comment by noahm** [comment]
#49 has been updated with more cleaned up data and a full pass on all songs currently unlockable, extra exclusive, golden league, etc.

Some old song data I'm not sure about:
- are the challenge charts from the 20th Anniversary Grand Finale event (Chaos terror-tech, MAX 360, etc) unlocked for everyone now or if you missed it are they unavailable?
- same question for the Challenge carnival plus event from last year

**Reply by ranatalus** [reply]
20th anniversary yes, those songs/charts should all be fully unlocked for everyone now. Challenge Carnival Plus no

**Comment by noahm** [comment]
#49 has been updated again with that unlock info corrected, all the "japan exclusive" flags corrected, and a bunch of jacket images added.
The only remaining songs without jackets are the ones without jackets on RemyWiki: (I can also take care of these soon if your friend doesn't have them all ready)
```
    'Bang Pad(Werk Mix)',
    'Crazy Hot',
    'Feidie',
    'Globe Glitter',
    'Good Looking',
    'GUILTY DIAMONDS',
    'HARD BRAIN',
    'Hella Deep',
    'High & Low',
    "LET'S CHECK YOUR LEVEL!",
    'Lightspeed',
    'LIKE A VAMPIRE',
    "MOVE! (We Keep It Movin')",
    'MUTEKI BUFFALO',
    'No Life Queen [DJ Command Remix]',
    'ONYX',
    'paparazzi',
    'PARTY ALL NIGHT(DJ KEN-BOW MIX)',
    "Poppin' Soda",
    'Rave Accelerator',
    'Run The Show',
    'Sword of Vengeance',
    'Taking It To The Sky (PLUS step)',
    'Uh-Oh',
    "Yuni's Nocturnal Days",
    'なだめスかし Negotiation',
    'シル・ヴ・プレジデント',
    'スカイクラッドの観測者',
    'スーパー戦湯ババンバーン',
    'ロキ(w/緒方恵美)',
    '勇猛無比',
    '最終鬼畜妹フランドール・Ｓ',
    '雑草魂なめんなよ！'
```

**Comment by noahm** [comment]
Ok, all of the above jackets are added (sourced from ZiV simfiles) and everything is merged in! Latest release is [deploying now to surge](https://github.com/noahm/DDRCardDraw/runs/3418073643?check_suite_focus=true), and downloadable offline copies are [in their usual spot](https://github.com/noahm/DDRCardDraw/releases/tag/v1.28.0)

If you spot anything off with the data just let me know! I know it's hard to sift through it manually and I'm planning a better (graphical) way to browse through all the flags and such to be added someday.

**Reply by ranatalus** [reply]
would it be possible to show somewhere in the settings area how many charts are in the pool based on current settings? that would be an easy way to at least know at a high level that things are/are not correct

**Reply by noahm** [reply]
Yeah that's a great idea! That shouldn't be too difficult to set up at all. I'll cook something like that up soon

**Reply by noahm** [reply]
@ranatalus got a quick version of that working in #50 ([preview build here](https://deploy-preview-50--ddrdraw.netlify.app/?game=a20plus)). Flip the new toggle in the options and the list of charts below will update in real-time as you adjust settings!

**Reply by noahm** [reply]
Following up to add: I went ahead and merged #50 so you can play around with it at the usual URL.
